### PR TITLE
lm_sensors: 3.4.0 -> 3.5.0

### PR DESCRIPTION
**Pull request by dtzWill**
Using releases posted on github,
they work and I'm not sure how those fedora URL's are found anyway.

###### Motivation for this change

https://github.com/lm-sensors/lm-sensors/blob/V3-5-0/CHANGES

https://github.com/lm-sensors/lm-sensors/releases/tag/V3-5-0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/os-specific/linux/lm-sensors/default.nix [hunk 1]
```diff
@@ -7,22 +7,21 @@ assert sensord -> rrdtool != null;
 
 stdenv.mkDerivation rec {
   name = "lm-sensors-${version}";
-  version = "3.4.0"; # don't forget to tweak fedoraproject mirror URL hash
+  version = "3.5.0"; # don't forget to tweak fedoraproject mirror URL hash
 
   src = fetchurl {
     urls = [
       # "http://dl.lm-sensors.org/lm-sensors/releases/lm_sensors-${version}.tar.bz2" # dead
```
Comment (c0bw3b) [inline pkgs/os-specific/linux/lm-sensors/default.nix]: Long dead so remove entirely

File: pkgs/os-specific/linux/lm-sensors/default.nix [hunk 2]
```diff
@@ -7,22 +7,21 @@ assert sensord -> rrdtool != null;
...
       # https://github.com/lm-sensors/lm-sensors/releases/... # only generated tarballs
-      "https://src.fedoraproject.org/repo/pkgs/lm_sensors/lm_sensors-${version}.tar.bz2/c03675ae9d43d60322110c679416901a/lm_sensors-${version}.tar.bz2"
+      #"https://src.fedoraproject.org/repo/pkgs/lm_sensors/lm_sensors-${version}.tar.bz2/c03675ae9d43d60322110c679416901a/lm_sensors-${version}.tar.bz2"
```
Comment (c0bw3b) [inline pkgs/os-specific/linux/lm-sensors/default.nix]: Probably it was taking the release commit from https://src.fedoraproject.org/rpms/lm_sensors/commits/
but Fedora is still building v3.4.0 for now: https://koji.fedoraproject.org/koji/packageinfo?packageID=809

File: pkgs/os-specific/linux/lm-sensors/default.nix [hunk 3]
```diff
@@ -7,22 +7,21 @@ assert sensord -> rrdtool != null;
...
+      https://github.com/lm-sensors/lm-sensors/archive/V3-5-0.tar.gz # only generated tarballs
```
Comment (c0bw3b) [inline pkgs/os-specific/linux/lm-sensors/default.nix]: That's the real upstream source from https://hwmon.wiki.kernel.org/lm_sensors and should come first
**But** this is not a stable URL for `fetchurl` (GitHub generated archive) so rather use `fetchzip` and compute the SHA hash over the unpacked content

**Comment by c0bw3b** [comment]
Basically it could be simplified to
```nix
  src = fetchzip {
    urls = "https://github.com/lm-sensors/lm-sensors/archive/V3-5-0.tar.gz"
    sha256 = "<somehash>";
  };
```

**Comment by dtzWill** [comment]
Thank you!

**Pull request merged**